CMakeLists.txt

```diff
@@ -69,23 +69,31 @@ OPTION(BUILD_WITH_CODEGEN_SUPPORT "Build the library with the Code Generation su
 
 OPTION(BUILD_WITH_MULTITHREADS "Build the library with the Multithreading support (required OpenMP)" OFF)
 IF(BUILD_WITH_MULTITHREADS)
-  SET(BUILD_WITH_NTHREADS "4" CACHE STRING "Number of threads")
+  INCLUDE(ProcessorCount)
+  ProcessorCount(NPROCESSOR)
+  IF(NOT NPROCESSOR EQUAL 0)
+    MATH(EXPR NPROCESSOR ${NPROCESSOR}/2)
+    SET(BUILD_WITH_NTHREADS ${NPROCESSOR} CACHE STRING "Number of threads")
+  ELSE()
+    SET(BUILD_WITH_NTHREADS "4" CACHE STRING "Number of threads") # Default setting
+  ENDIF()
+  
   string(REGEX MATCH "^[0-9]+$" BUILD_WITH_NTHREADS  ${BUILD_WITH_NTHREADS})
   IF(NOT BUILD_WITH_NTHREADS MATCHES "^[0-9]+$")
-    SET(BUILD_WITH_NTHREADS 4)
+    SET(BUILD_WITH_NTHREADS 4) # Default setting 4.
     MESSAGE(WARNING "the number of threads have to be an interger value, set to ${BUILD_WITH_NTHREADS}")
   ENDIF()
 ENDIF()
 
 IF(BUILD_WITH_CODEGEN_SUPPORT)
-  SET(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/find-external/CppAD/" ${CMAKE_MODULE_PATH})
-  ADD_PROJECT_DEPENDENCY(cppad)
-  ADD_PROJECT_DEPENDENCY(cppadcg)
+  ADD_PROJECT_DEPENDENCY(cppad 20200000.0 REQUIRED PKG_CONFIG_REQUIRES "cppad >= 20200000.0")
+  ADD_PROJECT_DEPENDENCY(cppadcg 2.4.1 REQUIRED PKG_CONFIG_REQUIRES "cppadcg >= 2.4.1")
```

Comment:
@nim65s Could you please verify that this is the proper way to add these dependencies?

Reply:
I updated this in #806 (9a378d448c77ad7283cd012fc11016853061f21b) :
- we can't get the version number for now in cppad / cppadcg in our [custom finders](https://github.com/jrl-umi3218/jrl-cmakemodules/blob/master/find-external/CppAD/Findcppad.cmake). So there is no use in defining them. If you really need to check their version number, and at the CMake level, we have to implement that in the submodule, so this calls for an issue or a PR there.
- I didn't put REQUIRED, which was a mistake of mine. So I guess you should rebase this PR on devel, and add this REQUIRED.
- PKG_CONFIG_REQUIRES would be useful only if someone was using crocoddyl.pc before, and was counting on finding a cppad requirement there. But I really hope that all projects dependent on crocoddyl are using its CMake exports right now, so there is no need for this transitional workaround.

Reply:
So if I understand correctly:

1) PKG_CONFIG_REQUIRES would continue to support for pkg-config
2) if we leave version numbers here, even if they are not useful, they are not harmful. So a future update of cmake module will be all that is needed to add version checking?

I would like to check the cppad and cppadcg versions, since there is no back-compatibility in cppadcg. 2.4.1 is compatible with cppad 2020, and not the versions before. And there are some API changes between cppad 2018 and cppad 2020

Reply:
1. PKG_CONFIG_REQUIRES idea is to deprecate support for pkg-config instead of brutally dropping it. But as we moved all our projects to pure CMake, nothing is tested with pkg-config anymore, so we'll have to drop its support at some point anyway. The size of the workarounds in the jrl-cmakemodule makes it really hard to maintain.
2. Is the compatibility between cppadcg 2.4.1 and cppad 2020 really an issue of yours, and not of cppadcg, or maybe package providers ?

Reply:
Very well, let's drop the pkg-config support then 
> Is the compatibility between cppadcg 2.4.1 and cppad 2020 really an issue of yours, and not of cppadcg, or maybe package providers ?

The problem is from cppadcg. Before cppadcg 2.4.1, there was no minimum version requirement in cppadcg for cppad. So, I did have to struggle a bit (pinocchio still asks for cppad>=2018). 
The minimal way to make sure of compatibility would be to only check for cppadcg>=2.4.1.

Reply:
I think this PR should solve this issue: https://github.com/jrl-umi3218/jrl-cmakemodules/pull/428

Reply:
Should I wait for https://github.com/jrl-umi3218/jrl-cmakemodules/pull/428 to be handled before merging this?

Reply:
Yes, I'll review that in the day.